site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+
+[source]
+----
+terraform init
+----
+
+It is always safe to run link:https://developer.hashicorp.com/terraform/cli/commands/init[init] multiple times, to bring the working directory up to date with changes in the configuration. Though subsequent runs may give errors, this command will never delete your existing configuration or state.
+
+=== Configuration Strategy
+
+Once we initialized the Terraform working directory we need to understand the different strategies to handle the configuration which we describe in detail in the following sections.
+
+[cols="2,4,4"]
+|===
+| Strategy | Advantages | Disadvantages
+
+| import | manage existing resources | some resources dont fully support the Terraform lifecycle
+| data source  | reference existing resources without having to manage them | requires a external process to manage these resources
+| create | create and then manage all resources through Terraform | bigger initial effort
+|===
+
+== Terraform Import
+
+Before we go in to Creating, Updating and Removing of resources we first look in to the link:https://developer.hashicorp.com/terraform/language/import[Import] and link:https://developer.hashicorp.com/terraform/language/data-sources[Data Source] functionality. If you have existing resources you want to manage via Terraform you need to import the current configuration to be able to manage it via Terraform.
+
+[NOTE.note]
+====
+The cleaner way is to create all resources through Terraform instead, but in certain scenarios you will have to use `Import` or `Data Source` and we go through a few examples in this and the next section.
+====
+//Additional input for documentation tbd. import cli https://developer.hashicorp.com/terraform/cli/import import config https://developer.hashicorp.com/terraform/language/import https://spacelift.io/blog/importing-exisiting-infrastructure-into-terraform  https://medium.com/swlh/importing-existing-infrastructure-into-terraform-a6ae168ad2bb https://stackoverflow.com/questions/47613926/import-all-resources-defined-in-tf-file https://www.bitslovers.com/terraform-import/)
+// The import and generate config command works and I've tested it, but the import configuration has a issue https://github.com/gpsinsight/terraform-provider-fusionauth/issues/214
+
+=== Importing The Default Tenant
+Importing the Default tenant is required if you want to manage it through Terraform as it's created during the initial deployment of FusionAuth. In addition, if you are managing FusionAuth application, it will always be part of the Default tenant. As well as there is no need to create a different tenant for creating applications.
+//knowledge source https://fusionauth.io/community/forum/topic/1725/what-are-the-pros-and-cons-of-using-the-default-tenant
+
+[NOTE.note]
+====
+With the usage of Terraform we suggest you to create your own tenant and applications related to your tenant instead of importing.
+====
```

Comment:
This is a bit of a headache to me to be honest, which why it is conflicting.

Importing the existing FusionAuth Tenant and Application is possible but it contradicts somewhat the typical Terraform workflow as its never created by terraform and a non deleteable resource too.

There are some benefits to it especially for a consistent setup, if you want terraform to setup or even further manage certain pieces of the configuration you have to import it, and e.g. can reset what ever a user has changed manually, or specifically ignore where users are allowed to do changes.

The alternative is either creating everything from scratch or using the existing setup as data source and managing the FusionAuth Application and Default Tenant outside of Terraform.

I did in the meantime setup something similar for our own fusionauth environment, and the best experience i had is with creating everything by Terraform including the Tenant. And therefore I personally would suggest not to import to Terraform, but then again it's all depending what you are willing to deal with and what your requirements are.

Reply:
Okay. I think if you put this into a paragraph/section and helped the user make the right choice for them (which you kinda did with a table), rather than having two pieces of conflicting advice right next to each other, that would be fine.